src/ui/features/onboarding/initialization.ts

```diff
@@ -1,34 +1,17 @@
 import { getCurrentUser } from 'src/shared/getCurrentUser';
 import { openOnboarding } from 'src/shared/openOnboarding';
-import { setUrlContext } from 'src/shared/setUrlContext';
-import { urlContext } from 'src/shared/UrlContext';
-import { UrlContextParam } from 'src/shared/types/UrlContext';
 import { OnboardingInterrupt } from './errors';
 
-function getAppMode({ hasExistingUser }: { hasExistingUser: boolean }) {
-  const hasOnboardingUrl = document.location.hash.startsWith('#/onboarding');
-  const isPopup = urlContext.windowType === 'popup';
-  return hasOnboardingUrl || (!isPopup && !hasExistingUser)
-    ? 'onboarding'
-    : 'wallet';
-}
-
 export async function maybeOpenOnboarding() {
   const hasExistingUser = Boolean(await getCurrentUser());
-  const isPopup = urlContext.windowType === 'popup';
+  const isOnboarding = document.location.hash.startsWith('#/onboarding');
 
-  if (isPopup && !hasExistingUser) {
+  if (!isOnboarding && !hasExistingUser) {
     openOnboarding();
+    await chrome.sidePanel.setPanelBehavior({
+      openPanelOnActionClick: false,
+    });
+    window.close();
     throw new OnboardingInterrupt();
   }
-
-  const appMode = getAppMode({ hasExistingUser });
```

Comment:
Need to check why this was done and is it safe to remove this?

Reply:
That was added during the task of making `esc` close dialogs. However, everything with dialog I was able to find, still works